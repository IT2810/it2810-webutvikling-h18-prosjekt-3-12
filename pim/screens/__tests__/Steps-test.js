import "react-native";
import React from "react";
import Steps from "../StepsScreen";
import renderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";
import Alert from "Alert";

//Snapshot testing
describe("Steps snapshot", () => {
  it("should render steps view correctly", () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(<Steps />);
    expect(result).toMatchSnapshot();
  });

  it("should render outer most view in steps view correctly", () => {
    //tests a snapshot view
    const tree = renderer.create(<Steps />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

//Unit testing
describe("Unit testing: tests functions", () => {
  const stepsContainer = renderer.create(<Steps />).getInstance();

  test('formated date is on format "Oct 02, 21:22"', () => {
    const formatedDate = stepsContainer.formatDate();

    //Month to match regex. 1 big letter, then 2 small letters
    const letters = formatedDate.substring(0, 3);
    expect(letters).toMatch(/[A-Z]{1}[a-z]{2}/);

    //Day to match regex. 2 numbers
    const day = formatedDate.substring(4, 6);
    expect(day).toMatch(/[1-9]{2}/);

    //Clock to match regex. 2 numbers, then colon, then 2 numbers
    const clock = formatedDate.substring(8, 14);
    expect(clock).toMatch(/[1-9]{2}[:]{1}[0-9]{2}/);
  });

  test("convert number of steps to distance (both in meters and km)", () => {
    //sets number of steps to 200 => should return distance in m
    stepsContainer.state.pastStepCount = 200;
    let meter_distance = (200 * 0.762).toFixed(0) + " m";
    expect(meter_distance).toMatch(stepsContainer.convertStepsToUnits());

    //sets number of steps to 2500 => should return distance in km
    stepsContainer.state.pastStepCount = 2500;
    let km_distance = ((2500 * 0.762) / 1000).toFixed(2) + " km";
    expect(stepsContainer.convertStepsToUnits()).toMatch(km_distance);
  });

  test("resets the current step count to 0", () => {
    stepsContainer.state.currentStepCount = 67;
    stepsContainer.onResetData();
    expect(stepsContainer.state.currentStepCount).toBe(0);
  });

  test("should handle the info icon press", () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(<Steps />);
    const mockFn = jest
      .fn()
      .mockName(
        result.props.children.props.children.props.children[1].props.children[1]
          .props.onPress
      );
    expect(mockFn).toBeTruthy();
  });
});
