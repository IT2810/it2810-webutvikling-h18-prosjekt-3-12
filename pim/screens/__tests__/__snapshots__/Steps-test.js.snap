// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps snapshot should render outer most view in steps view correctly 1`] = `
<View
  enabled={true}
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  />
</View>
`;

exports[`Steps snapshot should render steps view correctly 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <Component
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  >
    <ScrollViewMock
      contentContainerStyle={
        Object {
          "paddingTop": 30,
        }
      }
      style={
        Object {
          "backgroundColor": "#fff",
          "flex": 1,
        }
      }
    >
      <Component
        style={
          Object {
            "marginHorizontal": 40,
          }
        }
      >
        <Component
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "paddingBottom": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 20,
                "fontWeight": "bold",
              }
            }
          >
            Activity
          </Text>
          <Icon
            allowFontScaling={false}
            name="ios-walk"
            size={30}
            style={
              Object {
                "marginLeft": 10,
                "marginTop": -4,
              }
            }
          />
        </Component>
        <Component
          style={
            Object {
              "backgroundColor": "#E9F7FD",
              "borderColor": "#d6d7da",
              "borderRadius": 7,
              "borderWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 15,
              "overflow": "hidden",
              "padding": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(96,100,109, 1)",
                "fontSize": 17,
                "lineHeight": 24,
              }
            }
          >
            Current steps: 
            0
             steps
          </Text>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "marginTop": 5,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              name="ios-information-circle-outline"
              size={16}
              style={
                Object {
                  "color": "rgba(96,100,109, 1)",
                }
              }
            />
          </TouchableOpacity>
        </Component>
        <Component
          style={
            Object {
              "backgroundColor": "#E9F7FD",
              "borderColor": "#d6d7da",
              "borderRadius": 7,
              "borderWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 15,
              "overflow": "hidden",
              "padding": 20,
              "paddingBottom": 0,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(96,100,109, 1)",
                "fontSize": 17,
                "lineHeight": 24,
              }
            }
          >
            Past day: 
            0
             steps
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(96,100,109, 1)",
                "fontSize": 11,
                "marginVertical": 15,
              }
            }
          >
            Oct 18, 21:53
          </Text>
        </Component>
        <Component
          style={
            Object {
              "backgroundColor": "#E9F7FD",
              "borderColor": "#d6d7da",
              "borderRadius": 7,
              "borderWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 15,
              "overflow": "hidden",
              "padding": 20,
              "paddingBottom": 0,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(96,100,109, 1)",
                "fontSize": 17,
                "lineHeight": 24,
              }
            }
          >
            Distance: 
            0 m
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(96,100,109, 1)",
                "fontSize": 11,
                "marginVertical": 15,
              }
            }
          >
            Oct 18, 21:53
          </Text>
        </Component>
        <Component
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "marginTop": 0,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "#ff837c",
                  "borderRadius": 7,
                  "color": "white",
                  "fontSize": 12,
                  "lineHeight": 24,
                  "overflow": "hidden",
                  "padding": 5,
                }
              }
            >
              Reset data
            </Text>
          </TouchableOpacity>
        </Component>
        <Component
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "paddingTop": 40,
            }
          }
        >
          <Component
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "paddingBottom": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              Achivement
            </Text>
            <Icon
              allowFontScaling={false}
              name="ios-medal"
              size={20}
              style={
                Object {
                  "marginLeft": 10,
                  "marginTop": 2,
                }
              }
            />
          </Component>
          <Component
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Number of steps to achive pr day:
            </Text>
            <TextInput
              allowFontScaling={true}
              keyboardType="numeric"
              maxLength={5}
              onChangeText={[Function]}
              style={
                Object {
                  "borderColor": "#d6d7da",
                  "borderRadius": 7,
                  "borderWidth": 0.5,
                  "fontSize": 16,
                  "height": 40,
                  "marginLeft": 10,
                  "marginTop": -10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "width": 76,
                }
              }
            />
          </Component>
          <Progressbar
            currentStepCount={0}
            pastStepCount={0}
            stepsToAchive={0}
          />
        </Component>
      </Component>
    </ScrollViewMock>
  </Component>
</KeyboardAvoidingView>
`;
